src/vscode/commands/Repl.ts

```diff
@@ -66,28 +72,68 @@ export async function inlineEval(state: ExtensionState) {
     })
 }
 
-export async function attachRepl(state: ExtensionState, ctx: vscode.ExtensionContext) {
+export async function startReplAndAttach(state: ExtensionState, ctx: vscode.ExtensionContext) {
+    try {
+        const installedSlimeInfo = await installAndConfigureSlime(state)
+        const cwd = await getWorkspaceOrFilePath()
+        const cmd = vscode.workspace.getConfiguration('alive').swank.startupCommand
+        const attachToRunningRepl = (out: string) => {
+            swankOutputChannel.appendLine(out)
+            if (out.includes('Swank started at port: 4005')) {
+                attachRepl(state, ctx, { host: 'localhost', port: 4005 })
+            } else if (out.includes('Swank started at port')) {
+                attachRepl(state, ctx)
+            }
+        }
+        const handleDisconnect = (code: number, signal: string) => {
+            if (code === 0) {
+                vscode.window.showInformationMessage('REPL Detached')
+            } else {
+                disconnectAndClearChild(state)
+                vscode.window.showWarningMessage(`Unexpected REPL Detach (signal=${signal} code=${code})`)
+            }
+        }
+
+        if (state.child === undefined) {
+            state.child = spawn(cmd[0], cmd.slice(1), { cwd, env: getClSourceRegistryEnv(installedSlimeInfo.path, process.env) })
+            state.child.stdout?.setEncoding('utf-8').on('data', attachToRunningRepl)
+            state.child.stderr?.setEncoding('utf-8').on('data', attachToRunningRepl)
+            state.child
+                .on('close', handleDisconnect)
+                .on('exit', handleDisconnect)
```

Comment:
I think only 'exit' is necessary. Close is called when the streams are closed. Since we never keep the streams open, they'll only be closed when the process exits so handleDisconnect always ends up getting called twice.